frontend/app/src/components/StreamlitDialog/StreamlitDialog.test.tsx

```diff
@@ -38,13 +40,18 @@ describe("StreamlitDialog", () => {
     // Flush promises to give componentDidMount() a chance to run.
     await flushPromises()
 
-    setTimeout(() => {
-      expect(wrapper.find("button").at(1).is(":focus")).toBe(true)
-    }, 0)
+    await waitFor(() => {
+      const buttons = screen.getAllByRole("button")
+      const targetButton = buttons[1]
+
+      expect(targetButton).toHaveTextContent("Clear caches")
+      // eslint-disable-next-line testing-library/no-wait-for-multiple-assertions
+      expect(targetButton).toHaveFocus()
+    })
```

Comment:
Nit: Think better practice here is `await` & `findBy` to avoid this eslint error altogether:
```
const buttons = await screen.findAllByRole("button")
const targetButton = buttons[1]
expect(targetButton).toHaveTextContent("Clear caches")
expect(targetButton).toHaveFocus()
```

Reply:
Will keep this in mind, ty! Fixed!